dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/context/annotation/DubboClassPathBeanDefinitionScanner.java

```diff
@@ -67,4 +74,35 @@ public boolean checkCandidate(String beanName, BeanDefinition beanDefinition) th
         return super.checkCandidate(beanName, beanDefinition);
     }
 
+    @Override
+    public Set<BeanDefinition> findCandidateComponents(String basePackage) {
+        Set<BeanDefinition> beanDefinitions = beanDefinitionMap.get(basePackage);
+        if (CollectionUtils.isEmpty(beanDefinitions)) {
+            beanDefinitions = super.findCandidateComponents(basePackage);
+            beanDefinitionMap.put(basePackage, beanDefinitions);
+        }
+        return beanDefinitions;
+    }
+
+    /**
+     * Registers @Service Bean and Finds all BeanDefinitionHolders of @Service
+     *
+     * @param registry              BeanDefinitionRegistry
+     * @param beanNameGenerator     BeanNameGenerator
+     * @param basePackage           basePackage
+     * @return                      all BeanDefinitionHolders of @Service
+     */
+    public Set<BeanDefinitionHolder> doScan(BeanDefinitionRegistry registry, BeanNameGenerator beanNameGenerator, String basePackage) {
```

Comment:
Do not create new `doScan` method, there is already expose a `doScan` method abrove:
```java
    @Override
    public Set<BeanDefinitionHolder> doScan(String... basePackages) {
        return super.doScan(basePackages);
    }
```